### DISCUSSION TITLE
Calculating rupture length alongside mt solutions

### DISCUSSION THREAD

**Discussion by Donavin97** [nikosT/Gisola#10] [Ideas]
Hi @nikosT , I wonder if it could be possible to add in functions to perform more fault calculations alongside mt computations. For example fault rupture length and duration of rupture. 
If is is too far fetched, let me know and I'll close this discussion.
Thanks.

**Comment by nikosT** [comment]
Perhaps @esokos could answer that. Moreover, in case you have a code that calculates what you want, you can create a Pull Request (PR) with this enhancement and after check, it could be added to the main code fork.

**Comment by esokos** [comment]
These kind of calculations would need extended source inversion e.g. slip inversion. Gisola is not doing this at the moment, it is using point source approximation for the entire source to simplify things in automatic mode. Manual Isola on the other hand can be used in an extend source mode, treating the entire rupture as a series of point sources, in this way estimates of fault rupture length and duration can be derived. All this of course are mode important in the case of a large event (M>~6.5) and need very nice data around the source.

**Reply by Donavin97** [reply]
@esokos. Thanks for clarifying, unfortunately I am dealing with a sparce network, so rupture estimation won't work I suppose?

**Comment by esokos** [comment]
at least not for Gisola ... you may try manual Isola but it will need a strong event and enough data.

**Reply by nikosT** [reply]
@Donavin97, you could check on the manual ISOLA, here: [https://github.com/esokos/isola](https://github.com/esokos/isola)
and also the parallel version (but more oriented for automaticity purposes too) here: [https://github.com/nikosT/Gisola/tree/main/src/core](https://github.com/nikosT/Gisola/tree/main/src/core)
